src/util_libc.rs

```diff
@@ -76,29 +81,48 @@ pub fn sys_fill_exact(
 
 // A "weak" binding to a C function that may or may not be present at runtime.
 // Used for supporting newer OS features while still building on older systems.
-// F must be a function pointer of type `unsafe extern "C" fn`. Based off of the
-// weak! macro in libstd.
+// Based off of the DlsymWeak struct in libstd (src/sys/unix/weak.rs), except
+// that the caller must cast self.ptr() to a function pointer.
```

Comment:
Perhaps link directly to the struct?

```suggestion
// Based off of the DlsymWeak struct in libstd (https://github.com/rust-lang/rust/blob/1.61.0/library/std/src/sys/unix/weak.rs#L84),
// except that the caller must cast self.ptr() to a function pointer.
```